nibabel/tests/test_loadsave.py

```diff
@@ -135,3 +136,33 @@ def test_read_img_data_nifti():
             assert_array_equal(exp_offset, read_img_data(img_back))
             # Delete stuff that might hold onto file references
             del img, img_back, data_back
+
+
+def test_save():
+    with InTemporaryDirectory():
+        dataobj = np.ones((10, 10, 10), dtype=np.float16)
+        affine = np.eye(4, dtype=np.float32)
+        img = SpatialImage(dataobj, affine)
+
+        # The `save` method can raise one of many types of errors, but they are
+        # all subclasses of Exception. This attempt will fail because float16
+        # is not supported.
+        with assert_raises(Exception):
```

Comment:
Don't we know the error in this case though?

Reply:
You're right. I will make this more specific.